src/extensionApi.ts

```diff
@@ -79,6 +81,41 @@ export class CommandHandler {
         }
     }
 
+    public async debugServer(mode: string, context?: Protocol.ServerState): Promise<Protocol.StartServerResponse> {
+
+        const folderProject = await vscode.window.showOpenDialog({
+            canSelectFiles: true,
+            canSelectMany: false,
+            canSelectFolders: true,
+            openLabel: 'Open Project to debug'
+        } as vscode.OpenDialogOptions).then(
+            folder => {
+                if (folder != null && folder.length > 0) {
+                    vscode.commands.executeCommand('vscode.openFolder', folder[0]);
+                    return folder[0];
+                }
+            }
+        );
+
+        if (!folderProject) return;
+
+        if (vscode.extensions.getExtension('vscjava.vscode-java-debug') === undefined) {
```

Comment:
Should be done first in order not to ask user for useless informations

Reply:
That was my first thought, than i saw that the window got reloaded after the project was opened and I started thinking about it... below the two reasons it came to my mind:

1) The first I guess it's just a personal preference. Looking at the 2 different way a user can start the debug process:

click on the debug command -> ask for java extension -> open folder -> the process is stopped because vscode reloads the window (so the user get stopped at the second step and he has to start again)

click on the debug command -> open folder -> the window gets reloaded (the user is still at the beginning, so even if it start again the entire process he doesn't have the perception he needs to do all the steps again + he can follow all the steps without any interruption)

It's a subtle difference.

If the user already has the java extension installed, the debugging process will be the same in both situations, so we won't ask for useless infos.

click debug command -> choose folder -> start server -> start debug 

2) The second is ... what if in the future we will support a webserver that work with different programming language (not java)? In that case we wouldn't need the debugger java extension but a different one. So it would be better to see what's the project chosen by the user and suggest him the extension he has to install accordingly

BTW, at the moment i think the order is just personal taste, i'll switch them again.